src/Infrastructure.Dapper/Vault/Repositories/CipherRepository.cs

```diff
@@ -562,6 +562,57 @@ public async Task CreateAsync(IEnumerable<Cipher> ciphers, IEnumerable<Folder> f
                         }
                     }
 
+                    if (existingFolders.Any())
+                    {
+
+                        // 1. Create temp tables to bulk copy into.
+
+                        var sqlCreateTemp = @"
+                            SELECT TOP 0 *
+                            INTO #TempFolder
+                            FROM [dbo].[Folder]";
+
+                        using (var cmd = new SqlCommand(sqlCreateTemp, connection, transaction))
+                        {
+                            cmd.ExecuteNonQuery();
+                        }
+
+                        // 2. Bulk copy into temp tables.
+                        using (var bulkCopy = new SqlBulkCopy(connection, SqlBulkCopyOptions.KeepIdentity, transaction))
+                        {
+                            bulkCopy.DestinationTableName = "#TempFolder";
+                            var dataTable = BuildFoldersTable(bulkCopy, existingFolders);
+                            bulkCopy.WriteToServer(dataTable);
+                        }
```

Comment:
I don't follow why you need to do anything with existing folders. By definition they already exist in the database. If the cipher references the folderID appropriately, shouldn't it just work?

Reply:
is it part of the requirements to rename existing folder to whatever was imported? Why favor imported data over existing? that seems backwards. It also seems like there' a weird possibility to rename the folder/collection as the import progresses. The last cipher would get to set the name.

Reply:
Yes, is part of the requirement to update existing collections/folders with the imported data.